tests/v2/validation/provisioning/rke2/custom_cluster_test.go

```diff
@@ -78,41 +79,48 @@ func (c *CustomClusterProvisioningTestSuite) SetupSuite() {
 }
 
 func (c *CustomClusterProvisioningTestSuite) TestProvisioningRKE2CustomCluster() {
-	nodeRoles0 := []string{
-		"--etcd --controlplane --worker",
-	}
+	var etcdNode machinepools.NodeRoles
+	etcdNode.Etcd = true
 
-	nodeRoles1 := []string{
-		"--etcd --controlplane",
-		"--worker",
-	}
+	var controlPlaneNode machinepools.NodeRoles
+	controlPlaneNode.ControlPlane = true
 
-	nodeRoles2 := []string{
-		"--etcd",
-		"--controlplane",
-		"--worker",
-	}
+	var workerNode machinepools.NodeRoles
+	workerNode.Worker = true
+
+	var allRolesNode machinepools.NodeRoles
+	allRolesNode.Etcd = true
+	allRolesNode.ControlPlane = true
+	allRolesNode.Worker = true
+
+	var etcdControlPlaneNode machinepools.NodeRoles
+	etcdControlPlaneNode.Etcd = true
+	etcdControlPlaneNode.ControlPlane = true
+
+	var nodeRoles0 []machinepools.NodeRoles
+	nodeRoles0 = append(nodeRoles0, allRolesNode)
+
+	var nodeRoles1 []machinepools.NodeRoles
+	nodeRoles1 = append(nodeRoles1, etcdControlPlaneNode)
+	nodeRoles1 = append(nodeRoles1, workerNode)
+
+	var nodeRoles2 []machinepools.NodeRoles
+	nodeRoles2 = append(nodeRoles2, etcdNode)
+	nodeRoles2 = append(nodeRoles2, controlPlaneNode)
+	nodeRoles2 = append(nodeRoles2, workerNode)
```

Comment:
We can add more than one item at once with append

Reply:
yup I'm pretty sure you can do `nodeRoles2 = append(nodeRoles2, etcdNode, controlPlaneNode, workerNode)`